include/cudaq/Optimizer/InitAllDialects.h

```diff
@@ -10,13 +10,34 @@
 
 #include "cudaq/Optimizer/Dialect/CC/CCDialect.h"
 #include "cudaq/Optimizer/Dialect/Quake/QuakeDialect.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Complex/IR/Complex.h"
+#include "mlir/Dialect/ControlFlow/IR/ControlFlow.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/Math/IR/Math.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
 
 namespace cudaq {
 
-// Add all the MLIR dialects to the provided registry.
+// Add all required dialects to the provided registry.
 inline void registerAllDialects(mlir::DialectRegistry &registry) {
   // clang-format off
   registry.insert<
+    // MLIR dialects
+    mlir::AffineDialect,
+    mlir::arith::ArithDialect,
+    mlir::cf::ControlFlowDialect,
+    mlir::complex::ComplexDialect,
+    mlir::func::FuncDialect,
+    mlir::LLVM::LLVMDialect,
+    mlir::math::MathDialect,
+    mlir::memref::MemRefDialect,
+    mlir::scf::SCFDialect,
```

Comment:
We shouldn't be using Affine, MemRef,, or SCF. There may be some dusty deck tests that use those, but we don't need or want to use them.

Reply:
I agree, I think there are some really old test that actually use those. I just didn't wanted to go ahead removing them before getting your eyes on this (:

Reply:
This PR is a nice find, @boschmitt! So is the plan to make the Affine, MemRef, and SCF changes as part of this PR or to leave them for a future PR?

Reply:
I spoke with Eric offline (should have posted here). I think it would be best to leave this PR as-is, and work on removing those in a different one (I can do that too). I wanted to keep code/test changes to a minimum to facilitate review and making sure the different steps don't break other things. (I could also create a PR train base on this one.)